src/adm/handlers/moderation/awardBadge.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Given userId & award badge (Id and metadata).
+ *
+ */
+import { HTTPError } from 'fets';
+
+import client from 'util/client';
+
+/**
+ * Update user to write badgeId. Throws if user does not exist.
+ *
+ * @param userId
+ * @param badgeId
+ * @param badgeMetaData
+ */
+async function appendBadgeToList(
+  userId: string,
+  badgeId: string,
+  badgeMetaData: string
+) {
+  const now = new Date().toISOString();
+
+  try {
+    const {
+      body: { result: setbadgeIdResult },
+    } = await client.update({
+      index: 'users',
+      type: 'doc',
+      id: userId,
+      body: {
+        script: {
+          source: `
+            if (ctx._source.badges == null) {
+              ctx._source.badges = [];
+            }
+            ctx._source.badges.add(params.badge);
+          `,
+          params: {
+            badge: {
+              badgeId: badgeId,
+              badgeMetaData: badgeMetaData,
+              createdAt: now,
+              isDisplayed: true,
+              updatedAt: now,
+            },
+          },
+        },
+      },
+    });
+
+    /* istanbul ignore if */
+    if (setbadgeIdResult === 'noop') {
+      console.log(`Info: user ID ${userId} already has set the same badgeId.`);
```

Comment:
It seems that the script above do not check if the same badge has been awarded or not, and always performs `ctx._source.badges.add`.
In this case, this `console.log` here would never be executed.